daemon/api.go

```diff
@@ -586,78 +612,151 @@ func deleteOp(c *Command, r *http.Request) Response {
 }
 
 type packageInstruction struct {
+	progress.NullProgress
 	Action   string `json:"action"`
 	LeaveOld bool   `json:"leave_old"`
 	pkg      string
-	prog     progress.Meter
+	chin     chan interface{}
+	chout    chan interface{}
 }
 
-func (inst *packageInstruction) install() interface{} {
-	flags := snappy.DoInstallGC
-	if inst.LeaveOld {
-		flags = 0
+func (inst *packageInstruction) Agreed(intro, licenseFile string) bool {
+	inst.chout <- map[string]interface{}{
+		"license_ack_needed": true,
+		"license_intro":      intro,
+		"license_text":       licenseFile,
 	}
-	_, err := snappy.Install(inst.pkg, flags, inst.prog)
 
+	b, ok := (<-inst.chin).(bool)
```

Comment:
The idea of task input/output on itself feels fine, but this particular use makes me wonder. It means we're creating running logic pending for an indeterminate amount of time waiting for input from the user. Why don't we just askf or the installation request to come with the license confirmation, and drop the running task while that's not the case?